src/lib/parameters/parameters.cpp

```diff
@@ -758,6 +794,22 @@ param_set_internal(param_t param, const void *val, bool mark_saved, bool notify_
 	perf_end(param_set_perf);
 	param_unlock_writer();
 
+#if defined(CONFIG_PARAM_SERVER)
+
+	// If this is the parameter server, make sure that the client is updated
+	// TODO: Handle the possibility that this fails.
+	if (param_changed && remote_update) { param_server_set(param, val); }
+
+#endif
+
+#if defined(CONFIG_PARAM_CLIENT)
+
+	// If this is the parameter client, make sure that the server is updated
+	// TODO: Handle the possibility that this fails.
+	if (param_changed && remote_update) { param_client_set(param, val); }
+
```

Comment:
The implementation doesn't follow a client-server model, but a replica. For that the implementation should be symmetric and the same for both sides (with potentially minor differences, but not completely separate implementations).